src/TaskListItem.svelte

```diff
@@ -0,0 +1,63 @@
+<script>
+  import { createEventDispatcher } from "svelte";
+
+  import EmojiButton from "./EmojiButton.svelte";
+  import ToolTip from "./ToolTip.svelte";
+
+  export let task;
+
+  const dispatchEvent = createEventDispatcher();
+
+  function handleCompleteClick() {
+    dispatchEvent("complete", task);
+  }
+
+  function handleDeleteClick() {
+    dispatchEvent("delete", task);
+  }
+
+  // Ensure the store is getting updated with the updated text for the task:
+  function handleEdit() {
+    dispatchEvent("edit", task);
+  }
+</script>
+
+<li class:done={task.done}>
+  <div>
+    <ToolTip title="Click to toggle whether this task has been completed.">
+      <EmojiButton
+        on:click={handleCompleteClick}
+        symbol="🏆"
+        label="Completed"
+      />
+    </ToolTip>
+  </div>
+  <div>
+    <ToolTip title="Click to edit this task.">
+      <span
+        contenteditable="true"
+        bind:innerHTML={task.text}
+        on:input={handleEdit}
```

Comment:
FYI, `contenteditable` is janky as hell when you're trying to update a value in global state (which is what we're going to be doing with Redux). The alternative is to cook up an elaborate component that switches to an `<input>` when you click on it, but that also sucks. That `on:input` event pushes the value up to the Svelte store, but it does a store update _on every keystroke_. Just something to keep in mind. That being said, this isn't a super complex UI, so this is fine.